guides/document_lifecycle.md

```diff
@@ -0,0 +1,162 @@
+# Document Lifecycle
+This guide covers OpenSearch JavaScript Client API actions for Document Lifecycle. You'll learn how to create, read, update, and delete documents in your OpenSearch cluster. Whether you're new to OpenSearch or an experienced user, this guide provides the information you need to manage your document lifecycle effectively.
+
+## Setup
+Assuming you have OpenSearch running locally on port 9200, you can create a client instance
+with the following code:
+```javascript
+const { Client } = require('@opensearch-project/opensearch');
+
+const client = new Client({
+  node: 'http://localhost:9200',
+});
+```
+Next, create an index named `movies` with the default settings:
+```javascript 
+const index = 'movies';
+client.indices.exists({ index: index })
+  .then(exists => {
+    if (!exists) {
+      return client.indices.create({ index: index });
+    }
```

Comment:
The use of callback here makes it look more complicated and less readable.
```javascript
const exists = client.indices.exists({ index: index })
if (!exists) {
  return client.indices.create({ index: index });
}
```
Is simpler

Reply:
@nhtruong I tried to stick to one style. Let me fix it now?

Reply:
Ohhh, ok :) 👍